### PR TITLE
Improve error message for custom log field mismatch.

### PR DESCRIPTION
**Pull request by SolidWallOfCode**
Currently when there is a field mismatch the error message is terse and only note there is a problem. his changes the message to provide at least the last field processed to provide a slim hope of finding the root cause.

### CODE REVIEW

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/freebsd-github/1477/ for details.

**Comment by atsci** [comment]
Linux build *failed*! See https://ci.trafficserver.apache.org/job/linux-github/1370/ for details.

**Comment by atsci** [comment]
clang-analyzer build *successful*! See https://ci.trafficserver.apache.org/job/clang-analyzer-github/41/ for details.

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/freebsd-github/1478/ for details.

**Comment by atsci** [comment]
Linux build *failed*! See https://ci.trafficserver.apache.org/job/linux-github/1371/ for details.

**Comment by atsci** [comment]
clang-analyzer build *successful*! See https://ci.trafficserver.apache.org/job/clang-analyzer-github/42/ for details.

File: proxy/logging/LogBuffer.cc [hunk 1]
```diff
@@ -615,10 +618,14 @@ LogBuffer::resolve_custom_entry(LogFieldList *fieldlist, char *printf_str, char
...
 
         bytes_written += res;
-        field = fieldlist->next(field);
+        lastField = field;
+        field     = fieldlist->next(field);
       } else {
         Note("There are more field markers than fields;"
-             " cannot process log entry");
+             " cannot process log entry '%.*s'. Last field = '%s' printf_str='%s' pos=%d/%d count=%d alt_printf_str='%s'",
+             bytes_written, write_to, lastField == nullptr ? "*" : lastField->symbol(),
```
Comment (zwoop) [inline proxy/logging/LogBuffer.cc]: Hmmm, maybe it's completely unrealistic, but couldn't lastField be NULL here ?

Comment (SolidWallOfCode) [inline proxy/logging/LogBuffer.cc]: Yes. Which is why there is a check for that case.

**Pull request merged**